File: go/metric_event_data.go
// this file is @generated
package meteroid

import "time"

type MetricEventData struct {
	AggregationKey *string `json:"aggregation_key,omitempty"`

	AggregationType BillingMetricAggregateEnum `json:"aggregation_type"`

	Code string `json:"code"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	MetricId BillableMetricId `json:"metric_id"`

	Name string `json:"name"`

	ProductFamilyId ProductFamilyId `json:"product_family_id"`

	ProductId *ProductId `json:"product_id,omitempty"`

	SegmentationMatrix *MetricSegmentationMatrix `json:"segmentation_matrix,omitempty"`

	UnitConversionFactor *int32 `json:"unit_conversion_factor,omitempty"`

	UnitConversionRounding *UnitConversionRoundingEnum `json:"unit_conversion_rounding,omitempty"`

	UsageGroupKey *string `json:"usage_group_key,omitempty"`
}

File: python/meteroid/models/metric_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .billable_metric_id import BillableMetricId
from .billing_metric_aggregate_enum import BillingMetricAggregateEnum
from .metric_segmentation_matrix import MetricSegmentationMatrix
from .product_family_id import ProductFamilyId
from .product_id import ProductId
from .unit_conversion_rounding_enum import UnitConversionRoundingEnum


@dataclasses.dataclass
class MetricEventData(BaseModel):
    aggregation_type: BillingMetricAggregateEnum

    code: str

    created_at: datetime

    metric_id: BillableMetricId

    name: str

    product_family_id: ProductFamilyId

    aggregation_key: t.Optional[str] = None

    description: t.Optional[str] = None

    product_id: t.Optional[ProductId] = None

    segmentation_matrix: t.Optional[MetricSegmentationMatrix] = None

    unit_conversion_factor: t.Optional[int] = None

    unit_conversion_rounding: t.Optional[UnitConversionRoundingEnum] = None

    usage_group_key: t.Optional[str] = None
